### PR TITLE
Merchant deletes item

### LINKED ISSUE #79: User Story 53, Merchant deletes an item
```
As a merchant
When I visit my items page
And I click on a "delete" button or link for an item
I am returned to my items page
I see a flash message indicating this item is now deleted
I no longer see this item on the page
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #83

depends on #82

### PR DESCRIPTION
**Pull request by nicovigil1**
adds flash message to /dashboard/items after deleting an item as merchant, closes #79

### CODE REVIEW

File: spec/features/users/merchant/merchant_destroy_spec.rb [hunk 1]
```diff
@@ -1,8 +1,21 @@
...
-    context "I can destroy" do
-        it "different things" do
+    context "on /dashboard/items" do
+        it "can delete an item & see a flash message" do
+            merchant = create(:user, role: 1)
+            item = create(:item, user:merchant, thumbnail:"oldguy.jpeg")
+            allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(merchant)
+            visit dashboard_items_path
+            save_and_open_page
+            within "#item-#{item.id}" do
```
Comment (stoic-plus) [inline spec/features/users/merchant/merchant_destroy_spec.rb]: If there's a `save_and_open_page` it will do that when running tests
